router/router.go

```diff
@@ -315,6 +330,42 @@ func New(cfg *config.Configuration, rateConvertor *currency.RateConverter) (r *R
 	return r, nil
 }
 
+func applyBidderInfoConfigOverrides(bidderInfos config.BidderInfos, adaptersCfg map[string]config.Adapter) error {
+	for bidderName, bidderInfo := range bidderInfos {
+		if adapterCfg, exists := adaptersCfg[bidderName]; exists {
+			bidderInfo.Syncer = adapterCfg.Syncer.Override(bidderInfo.Syncer)
+
+			// comptability with legacy settings (if possible unambiguously)
+			if adapterCfg.UserSyncURL != "" {
+				if bidderInfo.Syncer == nil {
+					return fmt.Errorf("failed to apply legacy usersync_url setting for bidder %s, bidder does not define a user sync", bidderName)
+				}
+
+				endpointsCount := 0
+				if bidderInfo.Syncer.IFrame != nil {
+					bidderInfo.Syncer.IFrame.URL = adapterCfg.UserSyncURL
+					endpointsCount++
+				}
+				if bidderInfo.Syncer.Redirect != nil {
+					bidderInfo.Syncer.Redirect.URL = adapterCfg.UserSyncURL
+					endpointsCount++
+				}
+
+				if endpointsCount == 0 {
+					return fmt.Errorf("failed to apply legacy usersync_url setting for bidder %s, bidder does not define user sync endpoints", bidderName)
+				}
+
+				if endpointsCount > 1 {
+					return fmt.Errorf("failed to apply legacy usersync_url setting for bidder %s, bidder defines multiple user sync endpoints", bidderName)
+				}
+			}
+
+			bidderInfos[bidderName] = bidderInfo
```

Comment:
I had to stare at this for a few minutes. `bidderInfo` is a loop variable, so gets reused. However we are copying the `bidderInfo` struct into a map of them, so we copy normally. No issues here.

Reply:
Sorry for the confusion. Do you recommend a different way of writing it to be less confusing? .. or perhaps a comment might be helpful here?

Reply:
No, no changes, just I try to be alert when I see `bidderInfos[bidderName] = bidderInfo` where `bidderInfo` is a loop variable. :) Figured I would make a note here for other reviewers.